buildinfo/buildinfo.go

```diff
@@ -30,3 +34,45 @@ var Release = "no release"
 func IsStamped() bool {
 	return BuildTime != "{BUILD_TIME}"
 }
+
+type BuildInfo struct {
+	BuildTime string
+	HostName  string
+	GitCommit string
+	GitStatus string
+	Release   string
+}
+
+func New(
+	buildTime string,
+	hostName string,
+	gitCommit string,
+	gitStatus string,
+	release string,
+) *BuildInfo {
+	return &BuildInfo{
+		BuildTime: buildTime,
+		HostName:  hostName,
+		GitCommit: gitCommit,
+		GitStatus: gitStatus,
+		Release:   release,
+	}
+}
+
+func Current() *BuildInfo {
+	return New(
+		BuildTime,
+		HostName,
+		GitCommit,
+		GitStatus,
+		Release,
+	)
+}
+
+func (bi BuildInfo) HasRelease() bool {
+	return bi.Release != ""
```

Comment:
nit: the conditional logic in this function and the one below are so trivial, and not expected to grow. It doesn't feel worth it to me to write a unit test, and then have to add 40 lines of code to make this file mockable

Reply:
I created `HasRelease()`, because I think that we may have two conditions to check. (I should have added the other. My bad.). One condition is not empty. The other is not equal to our default (i.e., unstamped) value, `no release`. 

For `IsClean()`, I think it is more compact and pretty clear. For example, a client would write the following to check for the clean state:

```go
if bi.IsClean() {
  // awesome code
}
```

If we do not have it, then they write the following:

```go
if bi.GitStatus == buildinfo.CleanGitStatus {
  // awesome code
}
```

Also, in the off chance that we change how we are tracking the state, the client code does not need to be touched.

I am not sure when we/if we will ever need to mock this struct. Test code should be able to create an instance without too much trouble using the `buildinfo.New()`. Do you have a test scenario in mind?

Reply:
I don't mean to pick on this one too much, I'm more trying to give guidance on the whole, that we're a scrappy unfunded startup so I'm more worried about the failure more where we don't make it to market than the one where some conditional logic doesn't have a unit test.

What you have here is fine, just wanted to make this point :)